cmd/state-svc/internal/resolver/resolver.go

```diff
@@ -179,45 +176,31 @@ func (r *Resolver) RuntimeUsage(ctx context.Context, pid int, exec string, dimen
 
 // ReportRuntimeUsage is an alternate version of RuntimeUsage which meets the
 // needs of the ipc package.
-func (r *Resolver) ReportRuntimeUsage(ctx context.Context, pid, exec, projDir string) {
-	var (
-		headless  string
-		commitID  string
-		nameSpace string
-	)
-
-	proj, err := project.FromPath(projDir)
-	if err != nil {
-		multilog.Critical("Could not convert pid string to int in proxied runtime-usage report: %s", errs.JoinMessage(err))
-	} else {
-		headless = fmt.Sprintf("%t", proj.IsHeadless())
-		commitID = proj.CommitID()
-		nameSpace = proj.Namespace().String()
-	}
-
-	logging.Debug("Packing state-exec data for process %s running project %s from %s", pid, projDir, exec)
+func (r *Resolver) ReportRuntimeUsage(ctx context.Context, pid, exec, namespace, commit, headless string) error {
```

Comment:
```suggestion
func (r *Resolver) RuntimeUsageFromExec(ctx context.Context, pid, exec, namespace, commit, headless string) error {
```

Since the trigger is hard coded it makes sense to make the function specific to that. This function itself doesn't care about IPC, that just happens to be where you're using it from.

Does this still make sense as a separate function though? Why not call `RuntimeUsage()` directly?

Note you can refactor `RuntimeUsage()` to address both use-cases if that makes sense.